VsCode: Add loading spinner during autocomplete request on current line

ruediste opened the pull request:
## Description

Add loading spinner during autocomplete request on current line. Can be switched off in the vscode extension configuration.

#925 

## Checklist

- [x] The relevant docs, if any, have been updated or created

## Screenshots

https://github.com/user-attachments/assets/96a94935-2e9c-4755-b6c7-d3e5d08c3289


## Testing
Edit some code, watch the loading spinner.
Disable the spinner in the settings, validate the absence of the spinner.

sestinj reviewed (changes requested):
@ruediste This is super cool! Makes autocomplete feel more predictable. I made a few small adjustments because I think it's really important that this not be too noisy (removed background and underline, and set false by default)

There are two other things I'd probably like to see before merging:
- An animation that feels less noticeable—both the white color and the spinning are a lot of motion. I think that a light gray (HEX #888888) dot that maybe just very slowly pulses would do the trick
- Right now the spinner is flashing a lot when there's a quick completion. I think that we should only show the spinner if ~200ms have passed and the completion is still loading, and probably should show the spinner for a minimum of ~250ms

ruediste commented:
I added the delay and two icons, now with Synfig Studio, an open source 2D drawing/animation software. I did the dot, but the animation continues running in the background, so some times the dot pops up in it's large state, sometimes in it's small state. So I added a slow spinner as well. To me the spinner looks better. Just replace the `gif` files to see both versions.

Happy to adjust the icons if you have suggestions, I'm not much of a designer ...

ruediste commented:
The minimum display time is present as well.

sestinj commented:
@ruediste I'm going to close this for now—given the amount of attention we'd want to put into the UI for this in order to ship it, I just don't think we're going to be able to prioritize it in the near-term. I'm mostly trying to avoid a situation where we ship something without polishing, have a few users use it but share negative feedback, and then we have a maintenance burden we aren't able to prioritize.

If you have any thoughts to the contrary, I'm open to hearing. In the future if/when we get back to this, I'm confident this PR will still be a great reference!
